<launch>
  <arg name="paused" default="false"/>
  <arg name="rviz" default="false"/>
  <arg name="rqt" default="false"/>
  <arg name="static_spin" default="true"/>
  <arg name="spin_rate" default="0.0"/>
  <arg name="static_tilt" default="true"/>
  <arg name="tilt_angle" default="0.0"/>
  <arg name="static_pan" default="true"/>
  <arg name="pan_angle" default="0.0"/>
  <arg name="num_scans_to_collect" default="10"/>
  <arg name="output_file" default="~/sensor_params.yaml"/>

  <include file="$(find cylbot_gazebo)/launch/cylbot.launch">
	<arg name="paused" value="$(arg paused)"/>
	<arg name="rviz" value="$(arg rviz)"/>
	<arg name="rqt" value="$(arg rqt)"/>
	<arg name="world_name" value="$(find multisense_sensor_model)/worlds/sensor_model_params.world"/>
	<arg name="static_spin" value="$(arg static_spin)"/>
	<arg name="spin_rate" value="$(arg spin_rate)"/>
	<arg name="static_tilt" value="$(arg static_tilt)"/>
	<arg name="tilt_angle" value="$(arg tilt_angle)"/>
	<arg name="static_pan" value="$(arg static_pan)"/>
	<arg name="pan_angle" value="$(arg pan_angle)"/>
  </include>

  <node name="hokuyo_beam_params_em" pkg="multisense_sensor_model" type="hokuyo_beam_params_em" output="screen" required="true">
	<param name="num_scans_to_collect" value="$(arg num_scans_to_collect)" type="int"/>
	<param name="output_file" value="$(arg output_file)" type="str"/>
  </node>

</launch>
